### PR TITLE
Do not fail if publiccloud::azure::upload_img is called for already uploaded image

### PR DESCRIPTION
**Pull request by pdostal**
- Previous pull requests: #16785, #16813
- Example failure: [openqa.suse.de/t10889077](https://openqa.suse.de/t10889077)
- Related ticket: [poo#126914](https://progress.opensuse.org/issues/126914)
- Verification run: [pdostal-server.suse.cz/t4272](https://pdostal-server.suse.cz/t4272)

### CODE REVIEW

**Comment by clanig** [comment]
Thanks! LGTM.

File: lib/publiccloud/azure.pm [hunk 1]
```diff
@@ -82,7 +82,7 @@ sub find_img {
     eval {
         $image = decode_azure_json($json)->{name};
     };
-    record_info('INFO', "Cannot find image $name. Need to upload it.\n$@") if ($@);
+    record_info('IMG NOT-FOUND', "Cannot find image $name. Need to upload it.\n$@") if ($@);
```
Comment (grisu48) [inline lib/publiccloud/azure.pm:85]: NO NEED TO YELL!!1!

```suggestion
    record_info('Image not present', "Cannot find image $name. Need to upload it.\n$@") if ($@);
```

File: lib/publiccloud/azure.pm [hunk 2]
```diff
@@ -158,8 +158,8 @@ sub upload_img {
     $blobs =~ s/^\s+|\s+$//g;    # trim
     my @blobs = split(/\n/, $blobs);
     if (grep(/$img_name/, @blobs)) {
-        die("The upload_img() subroutine has been called even tho the $img_name blob exists. " .
-              "This means that ccloud/upload_image test module did not properly detected it.");
+        record_soft_fail('BLOB EXISTS', "The upload_img() subroutine has been called even tho the $img_name blob exists. " .
```
Comment (grisu48) [inline lib/publiccloud/azure.pm:161]: SAME AS ABOVE!!1!

```suggestion
        record_soft_fail('Image exists', "upload_img() called even though the $img_name blob exists.\nThis means that publiccloud/upload_image test module did not properly detect it.");
```

**Review by grisu48 — COMMENTED** [review]
Two comments, otherwise LGTM!

File: lib/publiccloud/azure.pm (hunk 1 above)
Comment (pdostal) [reply lib/publiccloud/azure.pm:85]: It's not yelling. We use uppercase letters in those boxes.

File: lib/publiccloud/azure.pm (hunk 2 above)
Comment (pdostal) [reply lib/publiccloud/azure.pm:161]: Same as above + it's not image but blob. See line `157`: https://github.com/os-autoinst/os-autoinst-distri-opensuse/blob/9a13199b5fc32abdcca93661a7a272e33a374dd7/lib/publiccloud/azure.pm#L157

File: lib/publiccloud/azure.pm (hunk 1 above)
Comment (grisu48) [reply lib/publiccloud/azure.pm:85]: Ok

File: lib/publiccloud/azure.pm (hunk 2 above)
Comment (grisu48) [reply lib/publiccloud/azure.pm:161]: Ok

**Review by grisu48 — APPROVED** [review]

**Pull request merged**